Incorrect construction of release JARs

A failure in the shading of bundled dependencies prevents us building release artefacts for 2.7.5b1. Tests after `developer-build` pass as expected, because they are using JARs directly.

The fault manifests itself directly when either the main or standalone JAR is used for execution in a trial build:
```
PS 275b1-trial> java -jar kit\jython-standalone.jar
Exception in thread "main" java.lang.NoClassDefFoundError: org/python/google/common/collect/MapMaker
        at org.python.core.ThreadStateMapping.<clinit>(ThreadStateMapping.java:31)
        at org.python.core.Py.<clinit>(Py.java:1789)
        at org.python.core.PySystemState.<clinit>(PySystemState.java:80)
        at org.python.util.jython.main(jython.java:593)
Caused by: java.lang.ClassNotFoundException: org.python.google.common.collect.MapMaker
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 4 more
```
(The path `.\kit` contains the JARs placed in `dist` by `ant full-build`.) A quick investigation shows that the contents of some dependency JARs are bundled into the fat JARs without the required change to their package names.

In the output from `ant -v jar-complete` (in a candidate `2.7.5b1`) we see:
```
   [jarjar] Fully-qualified classname does not match jar entry:
   [jarjar]   jar entry: META-INF/versions/9/module-info.class
   [jarjar]   class name: module-info.class
   [jarjar] Omitting META-INF/versions/9/module-info.class.
   [jarjar] adding entry com/google/common/util/concurrent/internal/InternalFutureFailureAccess.class
   [jarjar] adding entry com/google/common/util/concurrent/internal/InternalFutures.class
   [jarjar] adding directory com/google/common/annotations/
...
   [jarjar] adding entry com/google/common/annotations/Beta.class
...
```
where in `2.7.4` we see:
```
   [jarjar] adding entry org/python/google/common/util/concurrent/internal/InternalFutureFailureAccess.class
   [jarjar] adding entry org/python/google/common/util/concurrent/internal/InternalFutures.class
   [jarjar] adding directory com/google/common/annotations/
...
   [jarjar] adding directory org/python/google/common/annotations/
   [jarjar] adding entry org/python/google/common/annotations/Beta.class
...
```

So in 2.7.5 there is a failure connected with `META-INF/versions/9/module-info.class` in the Guava JAR, and this prevents the shading (renaming) of classes from the JAR, although the renaming _is_ applied to our code that calls the Guava classes. Later our code references the renamed class, but it cannot be found.

--- a/NEWS
+++ b/NEWS
@@ -43,6 +43,7 @@ New Features
       the registry or on the command line. The default (for now) is true.
 
 Jython 2.7.5a1 Bugs fixed
+    - [ GH-452 ] Incorrect construction of release JARs
     - [ GH-434 ] CVE-2023-34462(CVSS 6.5), CVE-2022-41915(CVSS 6.5) Jython 2.7.4 using a vulnerable Netty version 4.1.173.Final
     - [ GH-430 ] CVE-2026-5598 (CVSS 8.9) - Jython 2.7.4 using a vulnerable Bouncy Castle version 1.78.1
     - [ GH-416 ] Solve serialization backward incompatibility of oversized methods and functions

--- a/build.gradle
+++ b/build.gradle
@@ -192,6 +192,8 @@ dependencies {
     implementation 'org.ow2.asm:asm-commons:9.10.1'
     implementation 'org.ow2.asm:asm-util:9.10.1'
 
+    implementation 'com.google.guava:guava:33.6.0-jre'
+
     // Pin icu4j to latest 77.x until Jython moves to Java 11.
     // https://unicode-org.github.io/icu/download/77.html
     implementation 'com.ibm.icu:icu4j:77.1'

--- a/build.xml
+++ b/build.xml
@@ -861,7 +861,7 @@ The text for an official release would continue like ...
             <rule pattern="org.apache.commons.compress.**" result="org.python.apache.commons.compress.@1"/>
             <zipfileset src="extlibs/commons-io-2.22.0.jar" excludes="META-INF/versions/9/module-info.class"/>
             <rule pattern="org.apache.commons.io.**" result="org.python.apache.commons.io.@1"/>
-            <zipfileset src="extlibs/failureaccess-1.0.3.jar"/>
+            <zipfileset src="extlibs/failureaccess-1.0.3.jar" excludes="META-INF/**"/>
             <zipfileset src="extlibs/guava-33.6.0-jre.jar" excludes="META-INF/**"/>
             <rule pattern="com.google.**" result="org.python.google.@1"/>
             <zipfileset src="extlibs/icu4j-77.1.jar"/>

